public/pages/EditFeatures/containers/EditFeatures.tsx

```diff
@@ -376,6 +382,7 @@ export function EditFeatures(props: EditFeaturesProps) {
           handleSubmit(values, actions.setSubmitting)
         }
         validate={validateFeatures}
+        validateOnChange={false}
```

Comment:
This is the only change to fix formik validation issue for CategoryField

Reply:
Does this `validateOnChange` only control validating CategoryField or not? Just make sure we don't ignore validation for other parts which necessary.

Reply:
Yes. For `Features`, its `validateOnChange` is set to be true in [code here](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/da5ead30a8266f43ff5212d59f18ec75f4559a8b/public/pages/EditFeatures/containers/EditFeatures.tsx#L155). And when save button is clicked, it will do check for category field again in [handleSaveChanges](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/da5ead30a8266f43ff5212d59f18ec75f4559a8b/public/pages/EditFeatures/containers/EditFeatures.tsx#L278).

Reply:
sorry for reverting the change. has found it can not fix the issue completely. Has made such change to completely fix the issue: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/339/files#diff-b12f07c85b6d89dbdd055f0482e17c93168efdabb5b2d08ec6fde29ee7b6dfddR292